### PR TITLE
merge in nci dataset

### PR DESCRIPTION
**Pull request by miaecle**
This PR adds in:
tf_regression results on nci dataset;
join graphconvreg model into benchmark;
max_deg changes due to ferrocene molecules in nci, fixing #328 ;(performance for other graph conv model remained the same)
some minor modifications in the performance table(fix random seed for random sampling);
low-data benchmark script(stability issues remained to be solved, the same as #320 );

### CODE REVIEW

**Comment by miaecle** [comment]
@rbharath I have just closed #320 , which included the same thing in this PR. I am still trying to figure out the low-data stability issue(BTW, is random seed in keras already defined by np or tf?), but I guess it will be better to first merge in this part of nci dataset. Graph convolution regression on nci takes really long time, I am still waiting for the results, once done I will add into the table and `stable_results.csv`

**Comment by rbharath** [comment]
@miaecle Thanks for getting this in! Looking through the perf-table, the NCI scaffold split accuracy for the MT-NN looks very low (basically 0 accuracy on validation set), while accuracy is quite high with a random split. Do you have any intuition on what's happening there?

As a second question, do we have graph-conv numbers for NCI?

**Comment by rbharath** [comment]
@miaecle https://github.com/fchollet/keras/issues/2280 suggests that `np.random.seed()` and `tf.set_random_seed()` need to be called before `keras` is imported. Since `import deepchem as dc` imports `keras` under the hood, maybe one solution is to set both of these seeds before importing deepchem. For example

```
import numpy as np
np.random.seed(42)
import tensorflow as tf
tf.set_random_seed(42)
import deepchem as dc
```

**Comment by miaecle** [comment]
@rbharath I actually have run it twice, both showing stable 0.02 :(
graphconv results on nci will come out by tomorrow hopefully, I am thinking about extracting part of the data in nci to try scaffold on multitask, if the scaffold number for graphconv is still that low.
The random settings seem making sense, I will try it out!

**Comment by rbharath** [comment]
@miaecle Thanks for looking into the scaffold numbers. One sanity check we could do is to train a `LinearRegression` baseline using `sklearn` to see if simple linear models also fail to learn anything with scaffolds. One possibility is that almost everything might belong to a few scaffolds, meaning that test-compounds are very exotic looking. Worth looking into...

Also, note #329. I've activated `openbabel` imports for `GridFeaturizer`, but this seems to cause segfaults with `sklearn` sometimes. If you're seeing segfaults with sklearn, I recommend commenting out the `openbabel` import to avoid problems.

**Comment by miaecle** [comment]
@rbharath Thanks for the suggestion! I was also wondering if there are any unexpected problems in the dataset(since other datasets are all fine with the same splitting), I will try that! Also, thanks for the reminder of sklearn, I haven't played around with it before.

**Comment by rbharath** [comment]
@miaecle Let's try to get this code merged in soon so we have preliminary one-shot and NCI numbers in the benchmarks. We can try to figure out the instability issues in the Keras/graph-conv models in a separate PR.

**Comment by miaecle** [comment]
@rbharath tests passed. I agree with that it is a general issue for keras instability. Also there is going to be some update regarding nci, since results are strange for scaffold splitting and graphconv regression. For now, would you mind checking if everything is good for merging in?

**Comment by rbharath** [comment]
@miaecle Looks good. Thanks for getting this fixed quickly!

Going to go ahead and merge this in. We can add the low-data and NCI-graph-conv numbers once we figure out the instability issues.
